pkg/apiserver/apiserver.go

```diff
@@ -119,23 +120,17 @@ func NewAPIServer(clientConfig *rest.Config, coreClient kubernetes.Interface, kc
 
 // Run spawns a go routine that exits when apiserver is stopped.
 func (as *APIServer) Run() error {
-	const (
-		retries = 60
-	)
-	go as.server.PrepareRun().Run(as.stopCh)
-
-	for i := 0; i < retries; i++ {
-		ready, err := as.isReady()
+	go func() {
+		err := as.server.PrepareRun().Run(as.stopCh)
 		if err != nil {
-			return fmt.Errorf("checking readiness: %v", err)
+			klog.Warningf("API service stopped: %v", err)
 		}
+	}()
 
-		if ready {
-			return nil
-		}
-		time.Sleep(1 * time.Second)
-	}
-	return fmt.Errorf("timed out after %s waiting for api server to become healthy. Check the status by running `kubectl get apiservices v1alpha1.data.packaging.carvel.dev -o yaml`", retries*time.Second)
+	return wait.PollInfinite(time.Second, func() (bool, error) {
+		klog.Info("waiting for API service to become ready")
```

Comment:
would be nice to to add: "(hint: check the status via `kubectl get apiservices v1alpha1.data.packaging.carvel.dev -o yaml`)"